### ISSUE TITLE
[Proposel] Improving the speed and core functionality [v2]

### ISSUE THREAD

**Issue by cyrildewit** [cyrildewit/laravel-page-view-counter#37]
# Overview
<coming soon>

## Draft ideas

* Caching all the page views and use them to count (can by quite slow if there are 2 million views)
* Caching the page view count only
* Storing the page view count for various periods (configured in the config file or per model) in a table collumn that will be updated every with an artisan command.

**Ideas are very welcome!**

**Comment by usman22** [comment]
but for how much time we can store in cache if views are increasing in seconds?

**Comment by cyrildewit** [comment]
That is the problem we are facing.

**Comment by usman22** [comment]
so i think cache cannot be best option for this, because views are always increasing in seconds so we should not update cache again and again in seconds

**Comment by cyrildewit** [comment]
Maybe adding an extra counter in the JSON object that's done by incrementing. This could be retrieved in a few miliseconds.

Then scheduling a command that recounts this number every hour or day.

**Comment by usman22** [comment]
yes it is a good idea to store the counting of views separately instead of storing a new record on every request that will take a huge space in database as well, and will also be slower while fetching

**Comment by cyrildewit** [comment]
But by doing that, we can't ask for the total page views that are maked in the past 30 minutes, 2 days or 2 montsh. I can also make this optional for the users who wants this. functionality.

**Comment by usman22** [comment]
yeah this will be not possible to tell views for particular days, what if we make a separate table where we'll have one record for each page with a counter attribute, and using it in simple scenario to fetch just total views of a page but if someone asks for views of last month or last hour etc then use original table like we are using right now

**Comment by cyrildewit** [comment]
Little update!

I have been working on v2 in the past few days and changed the workflow a little bit.

I think that I found a sweet solution. When the user try to retrieve the page visits of a specific time range, the returned value will be stored in the cache with a time limit.
It is actually a pretty simple solution xD. My first idea was to store these returned values in a different database table, until I released that I could use the cache for this.

I wil also work on a new documentation with more info about the workflow and requirements.
